mico-core/src/main/java/io/github/ust/mico/core/resource/ServiceResource.java

```diff
@@ -382,6 +384,27 @@
             linkTo(methodOn(ServiceResource.class).getDependencyGraph(shortName, version)).withSelfRel()));
     }
 
+    /**
+     * Return yaml for a {@link MicoService} for the give shortname and version.
+     *
+     * @param shortName
+     * @param version
+     * @return
+     */
+    @GetMapping("/{" + PATH_VARIABLE_SHORT_NAME + "}/{" + PATH_VARIABLE_VERSION + "}" + "/yaml")
+    public ResponseEntity<Resource<MicoYamlResponseDTO>> getServiceYamlByShortNameAndVersion(@PathVariable(PATH_VARIABLE_SHORT_NAME) String shortName,
+                                                                                             @PathVariable(PATH_VARIABLE_VERSION) String version) {
+        String yaml;
+        try {
+            yaml = micoKubernetesClient.getYaml(getServiceFromDatabase(shortName, version));
+        } catch (KubernetesResourceException e) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Could not find any Service with name: '" + shortName);
```

Comment:
A `KubernetesResourceException` should not be mapped to `NOT_FOUND`. It should be `CONFLICT`. Use the message in the exception.
Proposal:
```
throw new ResponseStatusException(HttpStatus.CONFLICT, "Deployment of service '" + shortName + "' '" + version + "' has a conflict: " + e.getMessage());
```